client/client_update_test.go

```diff
@@ -606,3 +636,436 @@ func testUpdateRemoteNon200Error(t *testing.T, opts updateOpts, errExpected inte
 		}
 	}
 }
+
+// If there's no local cache, we go immediately to check the remote server for
+// root. Having extra spaces doesn't prevent it from validating during bootstrap,
+// but it will fail validation during update. So it will fail with or without
+// a force check (update for write).  If any of the other roles (except
+// timestamp, because there is no checksum for that) have an extra space, they
+// will also fail during update with the same error.
+func TestUpdateRemoteChecksumWrongNoLocalCache(t *testing.T) {
+	for _, role := range append(data.BaseRoles, delegationsWithNonEmptyMetadata...) {
+		testUpdateRemoteFileChecksumWrong(t, updateOpts{
+			serverHasNewData: false,
+			localCache:       false,
+			forWrite:         false,
+			role:             role,
+		}, role != data.CanonicalTimestampRole) // timestamp role should not fail
+
+		if role == data.CanonicalRootRole {
+			testUpdateRemoteFileChecksumWrong(t, updateOpts{
+				serverHasNewData: false,
+				localCache:       false,
+				forWrite:         true,
+				role:             role,
+			}, true)
+		}
+	}
+}
+
+// If there's is a local cache, and the remote server has the same data (except
+// corrupted), then we can just use our local cache.  So update succeeds (
+// with or without a force check (update for write))
+func TestUpdateRemoteChecksumWrongCanUseLocalCache(t *testing.T) {
+	for _, role := range append(data.BaseRoles, delegationsWithNonEmptyMetadata...) {
+		testUpdateRemoteFileChecksumWrong(t, updateOpts{
+			serverHasNewData: false,
+			localCache:       true,
+			forWrite:         false,
+			role:             role,
+		}, false)
+
+		if role == data.CanonicalRootRole {
+			testUpdateRemoteFileChecksumWrong(t, updateOpts{
+				serverHasNewData: false,
+				localCache:       true,
+				forWrite:         true,
+				role:             role,
+			}, false)
+		}
+	}
+}
+
+// If there's is a local cache, but the remote server has new data (some corrupted),
+// we go immediately to check the remote server for root.  Having
+// extra spaces doesn't prevent it from validating during bootstrap,
+// but it will fail validation during update. So it will fail with or without
+// a force check (update for write).  If any of the other roles (except
+// timestamp, because there is no checksum for that) have an extra space, they
```

Comment:
Small detail, but I think throughout all the comments, we should replace "have an extra space" for "have been modified in transit", or something that makes it more obvious that has been "malicious" intervention here.

This isn't immediately obvious what you mean by "have an extra space" until we reach `TestSwizzlerAddExtraSpace()`